public/js/check_username.js

```diff
@@ -1,15 +1,12 @@
-let config = {
-  apiKey: "AIzaSyBjMU3cwt-m8fD9Fx2VW1756re0PyRxDvg",
-  authDomain: "num-cl.firebaseapp.com",
-  databaseURL: "https://num-cl.firebaseio.com"
-};
-firebase.initializeApp(config);
```

Comment:
Mmmm, ya no es necesario esto en las functions? Si lo sacas por ser algo que parece secreto, [no lo es](https://stackoverflow.com/questions/37482366/is-it-safe-to-expose-firebase-apikey-to-the-public) por si acaso y en su momento era necesario para poder correr el comando de más abajo, `firebase.database()`, por ejemplo.

Reply:
No lo saqué porque fueran secretos! Lo que pasa es que eso "hardcodeaba" la base de datos que se usaba para validar si un nombre de usuario estaba ocupado o no a la de producción (incluso en development), entonces era muy incómodo debuggear. Lo que hice fue cambiar cómo se importan los scripts en `public/index.html` para traerlos desde `/__/` en vez de un CDN, e importar `/__/firebase/init.js` además de la app y la base de datos. Ese script lo que hace es llamar `initializeApp` con los datos del entorno en que está corriendo, entonces en Firebase configura lo que había antes, pero en development usa la configuración del emulador y podemos probarlo todo localmente. [Acá](https://firebase.google.com/docs/web/setup#from-hosting-urls_1) hay más info sobre cómo funcionan esas URLs para importar los scripts 🤩

Reply:
La raja!!